platform/editor.mimelookup/src/org/netbeans/api/editor/mimelookup/MimePath.java

```diff
@@ -131,7 +131,7 @@ public final class MimePath {
         "video" //NOI18N
     ));
     
-    private static final Map<String,Reference<MimePath>> string2mimePath = new HashMap<String,Reference<MimePath>>();
+    private static final Map<String,Reference<MimePath>> string2mimePath = new HashMap<>();
```

Comment:
Synchronization looks ok to me, but what about [`Collections.synchronizedMap`](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#synchronizedMap-java.util.Map-) and forgetting about those `synchronized` blocks?

Reply:
Valid point, I switched to `ConcurrentHashMap` which should be better than the suggested `synchronizedMap` (finer Locking). What do you think?

Reply:
@matthiasblaesing `ConcurrentHashMap` is more performant. I assume we won't be [storying nulls as values](https://www.baeldung.com/java-synchronizedmap-vs-concurrenthashmap#2-null-support) though.